apps/charterafrica/src/pages/opportunities/[slug].page.js

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+
+import HelpdeskPageContent from "@/charterafrica/components/HelpdeskPageContent";
+import HelpdeskPageHeader from "@/charterafrica/components/HelpdeskPageHeader";
+import PageInfo from "@/charterafrica/components/PageInfo";
+import { payload } from "@/charterafrica/lib";
+
+function Index({ blocks }) {
+  return blocks?.map((block) => {
+    switch (block?.slug) {
+      case "helpdesk-page-content":
+        return <HelpdeskPageContent {...block} key={block.slug} />;
+      case "helpdesk-page-header":
+        return <HelpdeskPageHeader {...block} key={block.slug} />;
+      case "page-info":
+        return <PageInfo {...block} key={block.slug} />;
+      default:
+        return null;
+    }
+  });
+}
+
+export async function getServerSideProps({
+  defaultLocale,
+  locale,
+  locales,
+  query,
+}) {
+  const { slug } = query;
+  const { docs: pages } = await payload.findPage(slug, {
+    locale,
+    fallbackLocale: defaultLocale,
+  });
+
+  if (!pages?.length) {
+    return { notFound: true };
+  }
+
+  const blocks =
+    pages[0].blocks?.map(({ blockType, ...other }) => ({
+      ...other,
+      slug: blockType,
+    })) ?? [];
+  const { menus } = await payload.findGlobal("navigation", {
```

Comment:
We definitely won't be doing this for every page added.

I think we will later consider something like `getGlobalProps()` to get footer navbar and menus props.

Your thoughts @kilemensi ?